src/wallet/wallet2.cpp

```diff
@@ -1569,13 +1569,20 @@ bool wallet2::add_address_book_row(const cryptonote::account_public_address &add
   a.m_payment_id = payment_id;
   a.m_description = description;
   
-  int key = (m_address_book.empty())? 0 : m_address_book.rbegin()->first;
-  bool r =  m_address_book.emplace(++key,a).second;
-  return r;
+  int old_size = m_address_book.size();
+  m_address_book.push_back(a);
+  if(m_address_book.size() == old_size+1)
```

Comment:
I'm getting a compilation warning on this, then a failure on an internal compiler error